### PR TITLE
Match the up attack indicator with the hit box

### PR DESCRIPTION
**Pull request by Zorgonia**
i know it didn't feel great last time, but it feels better this time because of two reasons
1. rocco changing it so that up attack polls means that you don't have to be precise
2. the reason why it didnt feel like attacks worked half the time (besides non polling issue) is cause the attack is part of a hitbox so if you throw it out before the enemy is within the range of the attack, you actually die based off the enemy hitting the collider for the hitbox. This is fixed by making the hitbox a trigger since all we care about is the length vector for the overlap box function anyways.

tl;dr it should feel a lot better now

### CODE REVIEW

**Comment by Zorgonia** [comment]
also might wanna just discuss this one tomorrow
